ynr/apps/sopn_parsing/helpers/pdf_helpers.py

```diff
@@ -123,6 +124,9 @@ def detect_top_page(self, document_heading):
 
         """
         similar_len = document_heading.intersection(self.get_page_heading_set())
-        if len(similar_len) / len(document_heading) < CONTINUATION_THRESHOLD:
+        if (
+            len(similar_len) / len(document_heading) < CONTINUATION_THRESHOLD
+            or not len(similar_len) / len(document_heading) == 1
```

Comment:
I'm not too clear about what the  extra conditional here is checking, can you explain?

Reply:
This checks whether the pages are identical. In the issue, the use case was a two page document for the same ward with identical headers.

Reply:
Ok, having thought about this more:

At the moment we're comparing if the _length_ of `A`'s heading is the same as the _length_ of `B`'s heading. This is likely be to be true a lot of the time, because we're taking the same slice of each heading.

I think we need to check if the _content_ is identical, and we can do this like so:

```diff
diff --git ynr/apps/sopn_parsing/helpers/pdf_helpers.py ynr/apps/sopn_parsing/helpers/pdf_helpers.py
index 0218d0ed..bad4e4df 100644
--- ynr/apps/sopn_parsing/helpers/pdf_helpers.py
+++ ynr/apps/sopn_parsing/helpers/pdf_helpers.py
@@ -124,9 +124,12 @@ class SOPNPageText:
 
         """
         similar_len = document_heading.intersection(self.get_page_heading_set())
+
+        headings_are_identical = similar_len == document_heading
+
         if (
             len(similar_len) / len(document_heading) < CONTINUATION_THRESHOLD
-            or not len(similar_len) / len(document_heading) == 1
+            or headings_are_identical
         ):
             self.is_top_page = False
         return self.is_top_page
```